services/QuillLMS/app/controllers/profiles_controller.rb

```diff
@@ -99,6 +87,29 @@ def staff
     render :staff
   end
 
+  private def student_exact_scores(user, activity_id, classroom_unit_id)
+    user_id = user.id
+    cache_key = "#{Student::EXACT_SCORES_CACHE_KEY}/#{user_id}/#{activity_id}/#{classroom_unit_id}"
+
+    Rails.cache.fetch(cache_key, expires_in: 8.hours) do
+      activity_sessions = ActivitySession
+        .includes(:concept_results, :activity, :unit)
+        .where(
+          user_id:,
+          activity_id:,
+          classroom_unit_id:,
+          state: ActivitySession::STATE_FINISHED
+        )
+
+      {
+        'sessions' => activity_sessions.map { |as| format_activity_session_for_tooltip(as, user) },
```

Comment:
@emilia-friedberg One thing to confirm with you, I'm explicitly only returning these keys, and not the original `ua`. Rails parameters can't be cached if they are unpermitted, so we'd have to be explicit about the keys either in the params or in the return hash.

Reply:
Yeah, we need to include at least the the `ua_id` and the `classroom_unit_id` at the top-level here. Right now we're sending: `ua_id`, `classroom_unit_id`, `unit_id`, and `activity_id`.